src/autoscaler/metricsforwarder/server/server.go

```diff
@@ -17,31 +17,84 @@ import (
 	"go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux"
 
 	"code.cloudfoundry.org/lager/v3"
+	"github.com/gorilla/mux"
 	"github.com/patrickmn/go-cache"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/tedsuo/ifrit"
 )
 
-func NewServer(logger lager.Logger, conf *config.Config, policyDB db.PolicyDB, credentials cred_helper.Credentials, allowedMetricCache cache.Cache, httpStatusCollector healthendpoint.HTTPStatusCollector, rateLimiter ratelimiter.Limiter) (ifrit.Runner, error) {
-	metricForwarder, err := forwarder.NewMetricForwarder(logger, conf)
+func NewServer(logger lager.Logger, conf *config.Config, policyDb db.PolicyDB, credentials cred_helper.Credentials, allowedMetricCache cache.Cache, httpStatusCollector healthendpoint.HTTPStatusCollector, rateLimiter ratelimiter.Limiter) (ifrit.Runner, error) {
+	metricForwarder, err := createMetricForwarder(logger, conf)
 	if err != nil {
-		logger.Error("failed-to-create-metricforwarder-server", err)
-		os.Exit(1)
+		return nil, fmt.Errorf("failed to create metric forwarder: %w", err)
 	}
 
-	mh := NewCustomMetricsHandler(logger, metricForwarder, policyDB, allowedMetricCache)
+	mh := NewCustomMetricsHandler(logger, *metricForwarder, policyDb, allowedMetricCache)
 	authenticator, err := auth.New(logger, credentials)
```

Comment:
We never get an error here. We can simplify the code here (at least that) and I would vote for do a small refactoring on the sinature of `createMetricForwarder`.